python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/__init__.py

```diff
@@ -158,7 +164,99 @@ def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return wrapped_client._unwrapped_converse(*args, **kwargs)  # type: ignore
+
+            with tracer.start_as_current_span("bedrock.converse") as span:
+                span.set_attribute(
+                    SpanAttributes.OPENINFERENCE_SPAN_KIND,
+                    OpenInferenceSpanKindValues.LLM.value,
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                if inference_config := kwargs.get("inferenceConfig"):
+                    invocation_parameters = safe_json_dumps(inference_config)
+                    _set_span_attribute(
+                        span, SpanAttributes.LLM_INVOCATION_PARAMETERS, invocation_parameters
+                    )
+
+                if system_prompts := kwargs.get("system", []):
+                    input_buffer = 1
+                    if system_messages := " ".join(
+                        prompt.get("text", "") for prompt in system_prompts
+                    ):
+                        span_prefix = f"{SpanAttributes.LLM_INPUT_MESSAGES}.{0}"
+                        _set_span_attribute(span, f"{span_prefix}.message.role", "system")
+                        _set_span_attribute(span, f"{span_prefix}.message.content", system_messages)
+                else:
+                    input_buffer = 0
+
+                if message_history := kwargs.get("messages"):
+                    for idx, request_msg in enumerate(message_history):
+                        # Currently only supports single text-based data
+                        # Future implementation can be extended to handle different media types
+                        if (
+                            isinstance(request_msg, dict)
+                            and (request_msg_role := request_msg.get("role"))
+                            and (request_msg_content := request_msg.get("content", [None])[0])
+                            and (request_msg_prompt := request_msg_content.get("text"))
+                        ):
+                            span_prefix = (
+                                f"{SpanAttributes.LLM_INPUT_MESSAGES}.{idx + input_buffer}"
+                            )
+                            _set_span_attribute(
+                                span, f"{span_prefix}.message.role", request_msg_role
+                            )
+                            _set_span_attribute(
+                                span, f"{span_prefix}.message.content", request_msg_prompt
+                            )
+
+                response = wrapped_client._unwrapped_converse(*args, **kwargs)
+                if (response_message := response.get("output", {}).get("message")) and (
+                    response_content := response_message.get("content", [None])[0]
```

Comment:
what's the structure of "content", in general it's assuming the model might be multimodal right? Is there a way we can guarantee that we're getting a text field or maybe even the first text field?

Given what we discussed, it might be worthwhile to get all the text fields in the content and concatenate them. We're more likely to get a report suggesting the organization of our span is bad than people noticing that data should be there, but isn't